### ISSUE TITLE
NeXus Ontology

### ISSUE THREAD

**Issue by benajamin** [nexusformat/NIAC#95]
An ontology for NeXus has been created and has its own [nexusformat repository](https://github.com/nexusformat/NeXusOntology). To be properly published, it should be added to a permanent namespace with a persistent URL.

The NIAC is requested to approve using the [PURL](https://purl.archive.org/help) namespace: http://purl.org/nexusformat/definitions/

Administration of PURL requires an email address (as a login ID and to receive messages).

**Comment by mkoennecke** [comment]
The purl.org/nexusformat/definitions link 404's on me.

**Comment by benajamin** [comment]
@mkoennecke Yes, it should 404 because it has not been published there yet. The proposal is asking permission, not forgiveness.

**Comment by benajamin** [comment]
Proposal is to approve the publishing the ontology via the [PURL](https://purl.archive.org/help) namespace: http://purl.org/nexusformat/definitions/

**Comment by benajamin** [comment]
@hgoerzig has this been implemented? Following the URL in a browser just gave me a 404 error.

**Comment by spc93** [comment]
Hi, I don't thik ths has been implemented yet. It needs someone with access to a NIAC functional email account to set up and manage the purl.org redirection.

**Comment by FreddieAkeroyd** [comment]
Hi @spc93  `nexus-purl@nexusformat.org` should send to yourself, Heike and Ben - we can swap it to `nexus-committee` at a later point. Are you able to do the initial registration?

**Comment by Tanmay2028** [comment]
@benajamin Hi. I am Tanmay from FAU Erlangen Nürnberg. I am currently working on the OSCARS project. The NeXusOntology is still not available on this URL: http://purl.org/nexusformat/definitions/

Could you please check this once?

**Comment by lukaspie** [comment]
@phyy-nx this is the relevant issue regarding the namespace for the NeXusOntology. In short, what needs to happen for this to work:
- someone from the NIAC needs to register a [PURL](https://purl.archive.org/help) account (learn more on their [help page](https://purl.archive.org/help)) using a NIAC functional email account (`nexus-purl@nexusformat.org`?).
- Afterwards, the domain `nexusformat/definitions/` can be registered.
- Then, one needs to add the target URL where the OWL file is hosted, path can be from the release of the ontology: `https://github.com/nexusformat/NeXusOntology/releases/latest/download/NeXusOntology.owl` or similar (requires a release of the ontology itself).

Relevant discussion: https://github.com/nexusformat/NeXusOntology/issues/6

Please also notice our updates to the ontology itself in https://github.com/nexusformat/NeXusOntology/pull/7.

cc @mkuehbach

**Comment by hgoerzig** [comment]
It should work now. 
There are no releases in the ontology github managed, so I this https://raw.githubusercontent.com/nexusformat/NeXusOntology/refs/heads/main/ontology/NeXusOntology.owl as target.
Let me know if it is fine.

**Comment by lukaspie** [comment]
> It should work now. There are no releases in the ontology github managed, so I this [nexusformat/NeXusOntology@`refs`/heads/main/ontology/NeXusOntology.owl (raw)](https://raw.githubusercontent.com/nexusformat/NeXusOntology/refs/heads/main/ontology/NeXusOntology.owl) as target. Let me know if it is fine.

Thanks, the PURL link resolves now. There should still be a discussion for how this can be updated (release cycles aligned with NXDL releases?). Should be discussed in a NIAC telco ideally.

**Comment by phyy-nx** [comment]
I think in the last NIAC, @PeterC-DLS said he'd add updating the ontology to his checklist for releases.  @PeterC-DLS can you confirm?  Though it might be useful to have someone else pitch in on this when it comes up.

**Comment by PeterC-DLS** [comment]
I can add a note to trigger an update of the ontology but someone has to add in whatever CI/CD to do so.

**Comment by hgoerzig** [comment]
Can you please add someone more to the nexus-purl@nexusformat.org mailing list e.g. @paulmillar ?

**Comment by FreddieAkeroyd** [comment]
hi @hgoerzig `nexus-purl` is currently an alias rather than a mailing list, so at the moment only I can update names. I'm happy to do this but going forward
- should i now make `nexus-purl` a proper mailing list so more people can edit membership via the mailing list web interface, or
- did we want `nexus-purl` to now forward all messages to the existing `nexus-committee` list

**Comment by hgoerzig** [comment]
@FreddieAkeroyd nexus-purl has only been created for registering at archive.org and creating a persistent url. It is only needed for logging in and retrieving the password.

To some of you I have already mentioned that purl.org is awful to manage and it experiences frequent downtimes. Therefore, I have had a look at w3id.org persistent urls. They are using github for configuring the redirects and it works with pull requests on their repo.

We have used purl.org a few years ago because we did not know better. But maybe we should change?
I have created a fork of the repo and added a folder for nexusformat. I am happy to change the contact details and the redirects might have to be further adapted. You can have a look here:
https://github.com/hgoerzig/w3id.org/tree/master/nexusformat
